doc/user/content/sql/select.md

```diff
@@ -53,37 +53,41 @@ have on Materialize.
 
 ### Creating materialized views
 
-Creating a materialized view generates a persistent dataflow, which has a
+Creating a [materialized view](/sql/create-materialized-view) generates a persistent dataflow, which has a
 different performance profile from performing a `SELECT` in an RDBMS.
 
 A materialized view has resource and latency costs that should
 be carefully considered depending on its main usage. Materialize must maintain
 the results of the query in durable storage, but often it must also maintain
 additional intermediate state.
 
-### Reading from indexed relations
+### Creating indexes
+
+Creating an [index](/sql/create-index) also generates a persistent dataflow. The difference from a materialized view is that the results are maintained in memory rather than on persistent storage. This allows ad hoc queries to perform efficient point-lookups in indexes.
+
+### Ad hoc queries
+
+An ad hoc query (a.k.a. one-off `SELECT`) simply performs the query once and returns the results.
+
+#### Reading from indexed relations
 
 Performing a `SELECT` on an **indexed** source, view or materialized view is
 Materialize's ideal operation. When Materialize receives such a `SELECT` query,
 it quickly returns the maintained results from memory.
-
 Materialize also quickly returns results for queries that only filter, project,
-and re-order results.
+and re-order data that is maintained by an index.
 
-### Ad hoc queries
+#### Queries that will launch an ephemeral dataflow
 
-Queries over non-materialized views will create an ephemeral dataflow to compute
+Queries that can't simply read out from an index will create an ephemeral dataflow to compute
 the results. These dataflows are bound to the active [cluster](/get-started/key-concepts#clusters),
  which you can change using:
 
 ```sql
 SET cluster = <cluster name>;
 ```
 
-Performing a `SELECT` query that does not directly read out of a dataflow
-requires Materialize to evaluate your query. Materialize will construct a
-temporary dataflow to materialize your query, and remove the dataflow as soon as
-it returns the query results to you.
+Materialize will remove the dataflow as soon as it has returned the query results to you.
```

Comment:
This whole section could go under the main ad-hoc queries blurb.

Reply:
Yes! Sorry for the confusion, that's exactly what I meant!

Reply:
No problem! Done!